pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -729,7 +729,7 @@ func (s *ServiceEntryStore) GetProxyWorkloadLabels(proxy *model.Proxy) (labels.C
 	s.storeMutex.RLock()
 	defer s.storeMutex.RUnlock()
 
-	out := make(labels.Collection, 0)
+	out := make(labels.Collection, 0, len(proxy.IPAddresses))
```

Comment:
this is double loop, not quite right

Reply:
Ack. Missed this to push due to rebase. Done